### DISCUSSION TITLE
Like to integrate SlickGrid but...

### DISCUSSION THREAD

**Discussion by RenRoe** [6pac/SlickGrid#913] [Q&A] [answered]
Hello everyone, I hope I can get some answers to my questions here.

I need an image of the Microsoft Forms FlexGrid in the browser that is as accurate as possible because I am writing an application that reads most of the properties and dynamic data of the FlexGrid, creates a JSON from it and transfers it to the browser session via WebSockets. The design of the FlexGrid with the Slick Grid should be reproduced as accurately as possible in the browser.

To do this, I need functions that I either haven't found yet or require an expansion of the slick grid.

This includes:
     
1) Different row heights of data rows and header rows in the grid
If possible, the contents should be calculated in the background and the formatting of the cell contents should be taken into account. This calculated height of a line should be cached by the line div or the cell divs.

2) For each(!) UI element of the grid, events should exist to define the style, e.g. for viewports, and thus the Alpine style can be overruled. For normal websites this is certainly sufficient if static styles are set in the CSS, but for my purposes I need more flexibility.
(Background colors, Grid Colors and Grid lines width, Selection colors...) 

3) Automatic expanding of the last column to the right edge of viewport, when there is room to do so (ExpandLastColumn feature of FlexGrid)

I would like to participate in the development of these features, but I would like to know how you maintain the typescripts and what tools you use to export the js files. I live in the Windows world, so Linux tools are of little use to me.

**Comment by ghiscoding** [comment, marked as answer]
1. not possible, only Row Detail plugin exists which opens a panel by again with fixed height... searching on the internet you would have found plenty of Stack Overflow questions asking about the fixed row height
2. the new Alpine Theme is just a sample, just override it or create a new one from scratch (like I did when creating the Alpine Theme)
3. there is the window autoresize plugin that you can take a look at 

The project is now in TypeScript and compiled by esbuild to be able to export into 3 targets (iife, csm & esm), it's all explained in the [5.0 Announcement](https://github.com/6pac/SlickGrid/wiki/Major-version-5.0-%E2%80%90-ES6-ESM-and-TypeScript-Support), so I assumed you didn't read it, you'll get more info by reading it. I'm also myself strictly using Windows but regardless, all these tools are not bound to any specific OS, all you need is Node18

**Reply by RenRoe** [reply]
A quick look at the grid's jscript shows that the total height of the grid is calculated very easily using a formula. A difference algorithm when adding/removing grid rows might be feasible. Then finding the position of a line would be a very quick addition of the previous lines. But that would be an area for experimentation. It would also be possible to expand the last visible column if you add the column widths and take possible free space plus possible shown vertical scroll area on the right into account. I did all that stuff, when I have built a grid from scratch for windows forms with device contexts, to calculate if scroll areas might be visible or not in vertical or horizontal direction.

**Reply by RenRoe** [reply]
AutoSize is fine, but as far as I have seen, it is setting container height of the whole grid. Whenever there are different fonts with different font heights in a cell, or there is word wrap, or you have images part of a cell, the cell height is dynamic. It should be taken into account. With a function autosizeRow(row) and calling an event for each visible cell in the row you might get a value for optimize row height, and you simply sub the old row height and add the new row height to the total row height.

**Reply by ghiscoding** [reply]
I mentioned that you should search on the internet for SlickGrid dynamic row height, it doesn't look like you did, because you would have found this great Stack Overflow answering this question
- Stack Overflow answer: https://stackoverflow.com/a/29399927/1212166
- she also wrote a blog post with demos: https://violet313.github.io/

We actually used this approach to create the Row Detail Panel plugin, it is not easily doable to modify SlickGrid to support it though and we will not spend time doing it either. This is an Open Source project, if you want something done, the best would be to contribute to the project. There's already too much of my spare time that went into this project for the past 5 years already, it's time to let others contribute

**Comment by RenRoe** [comment]
I am sorry for it. You are right. The grid is fine, I like the approach, and such an expansion is more easily made just for my project, without the aspects of community updates, and compatibility reasons and code review. I will praise your excellent work in detail.
